File: Modules/ServerTypes/EmployeeManagement/EmployeesForm.ts
import { StringEditor, EnumEditor, MultipleImageUploadEditor, DateEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EmployeeGender } from "../Modules/EmployeeManagement.Employees.EmployeeGender";

export interface EmployeesForm {
    FirstName: StringEditor;
    LastName: StringEditor;
    Gender: EnumEditor;
    Certificates: MultipleImageUploadEditor;
    BirthDate: DateEditor;
    IsActive: BooleanEditor;
}

export class EmployeesForm extends PrefixedContext {
    static readonly formKey = 'EmployeeManagement.Employees';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeesForm.init)  {
            EmployeesForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = MultipleImageUploadEditor;
            var w3 = DateEditor;
            var w4 = BooleanEditor;

            initFormType(EmployeesForm, [
                'FirstName', w0,
                'LastName', w0,
                'Gender', w1,
                'Certificates', w2,
                'BirthDate', w3,
                'IsActive', w4
            ]);
        }
    }
}

[EmployeeGender]; // referenced types

File: Modules/EmployeeManagement/Employees/EmployeesForm.cs
using Serenity.ComponentModel;
using System;
using MySereneApp.Modules.EmployeeManagement.Employees;

namespace MySereneApp.EmployeeManagement.Forms;

[FormScript("EmployeeManagement.Employees")]
// [BasedOnRow(typeof(EmployeesRow), CheckNames = true)]
public class EmployeesForm
{
    [DisplayName("Forms.Employee.FirstName")]
    public string FirstName { get; set; }

    [DisplayName("Forms.Employee.LastName")]
    public string LastName { get; set; }

    [DisplayName("Forms.Employee.Gender")]
    public EmployeeGender Gender { get; set; }

    [MultipleFileUploadEditor]
    public List<string> Certificates { get; set; }

    [DisplayName("Forms.Employee.BirthDate")]
    public DateTime BirthDate { get; set; }

    public bool IsActive { get; set; }
}